src/app/clusters/operational-state-server/operational-state-server.cpp

```diff
@@ -0,0 +1,413 @@
+
+    // set operational error
+    SetOperationalState(currentOperationalState);
+
+    // copy current operational error
+    mOperationalError = currentErrState;
+
+    // return operational error
+    response.commandResponseState.errorStateID = static_cast<OperationalState::ErrorStateEnum>(mOperationalError.ErrorStateID);
+    response.commandResponseState.errorStateLabel.SetNonNull(mOperationalError.ErrorStateLabel,
+                                                             strlen(mOperationalError.ErrorStateLabel));
+    response.commandResponseState.errorStateDetails.Emplace(mOperationalError.ErrorStateDetails,
+                                                            strlen(mOperationalError.ErrorStateDetails));
+
+    ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+}
+
+void OperationalStateServer::HandleStartState(HandlerContext & ctx, const Commands::Start::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "OperationalState: HandleStartState");
+    Commands::OperationalCommandResponse::Type response;
+    OperationalErrorStateStruct currentErrState;
+    OperationalStateStruct currentOperationalState;
+
+    // copy current operational state
+    currentOperationalState = mOperationalState;
+    // copy current operational error
+    currentErrState = mOperationalError;
+
+    // callback
+    mDelegate->HandleStartState(currentOperationalState, currentErrState);
+
+    // set operational error
+    SetOperationalState(currentOperationalState);
+
+    // copy current operational error
+    mOperationalError = currentErrState;
+
+    // return operational error
+    response.commandResponseState.errorStateID = static_cast<OperationalState::ErrorStateEnum>(mOperationalError.ErrorStateID);
+    response.commandResponseState.errorStateLabel.SetNonNull(mOperationalError.ErrorStateLabel,
+                                                             strlen(mOperationalError.ErrorStateLabel));
+    response.commandResponseState.errorStateDetails.Emplace(mOperationalError.ErrorStateDetails,
+                                                            strlen(mOperationalError.ErrorStateDetails));
+
+    ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+}
+
+void OperationalStateServer::HandleStopState(HandlerContext & ctx, const Commands::Stop::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "OperationalState: HandleStopState");
+    Commands::OperationalCommandResponse::Type response;
+    OperationalErrorStateStruct currentErrState;
+    OperationalStateStruct currentOperationalState;
+
+    // copy current operational state
+    currentOperationalState = mOperationalState;
+    // copy current operational error
+    currentErrState = mOperationalError;
+
+    // callback
+    mDelegate->HandleStopState(currentOperationalState, currentErrState);
+
+    // set operational error
+    SetOperationalState(currentOperationalState);
+
+    // copy current operational error
+    mOperationalError = currentErrState;
```

Comment:
As per https://github.com/project-chip/connectedhomeip/pull/26971#discussion_r1212389891 please check whether this in fact needs to happen for every command.